fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/NoAnalysis.kt

```diff
@@ -0,0 +1,242 @@
+) {
+    ReviewQualityCheckCard(
+        modifier = Modifier.fillMaxWidth(),
+    ) {
+        var isAnalyzing by remember { mutableStateOf(false) }
+
+        Image(
+            painter = painterResource(id = R.drawable.shopping_no_analysis),
+            contentDescription = null,
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(90.dp)
+                .padding(start = 6.dp, top = 8.dp, end = 9.dp, bottom = 10.dp),
+        )
+
+        Spacer(Modifier.height(10.dp))
+
+        if (isAnalyzing) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                LoadingSpinner()
+
+                Spacer(modifier = Modifier.width(8.dp))
+
+                Text(
+                    text = stringResource(id = R.string.review_quality_check_analysis_in_progress_warning_title),
+                    style = FirefoxTheme.typography.headline8,
+                    color = FirefoxTheme.colors.textPrimary,
+                    modifier = Modifier.fillMaxWidth(),
+                )
+            }
+
+            Spacer(Modifier.height(8.dp))
+
+            Text(
+                text = stringResource(id = R.string.review_quality_check_analysis_in_progress_warning_body),
+                style = FirefoxTheme.typography.body2,
+                color = FirefoxTheme.colors.textSecondary,
+                modifier = Modifier.fillMaxWidth(),
+            )
+        } else {
+            Text(
+                text = stringResource(id = R.string.review_quality_check_no_analysis_title),
+                style = FirefoxTheme.typography.headline8,
+                color = FirefoxTheme.colors.textPrimary,
+                modifier = Modifier.fillMaxWidth(),
+            )
+
+            Spacer(Modifier.height(10.dp))
+
+            Text(
+                text = stringResource(id = R.string.review_quality_check_no_analysis_body),
+                style = FirefoxTheme.typography.body2,
+                color = FirefoxTheme.colors.textSecondary,
+                modifier = Modifier.fillMaxWidth(),
+            )
+
+            Spacer(Modifier.height(8.dp))
+
+            TextButton(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .background(
+                        color = FirefoxTheme.colors.actionPrimary,
+                        shape = RoundedCornerShape(4.dp),
+                    ),
+                onClick = {
+                    isAnalyzing = true
+                    onAnalyzeClick()
+                },
+            ) {
+                Text(
+                    text = stringResource(id = R.string.review_quality_check_no_analysis_link),
+                    color = FirefoxTheme.colors.textOnColorPrimary,
+                    style = FirefoxTheme.typography.headline8,
+                )
+            }
+        }
+    }
+}
+
+@Suppress("MagicNumber")
+@Composable
+private fun LoadingSpinner(
```

Comment:
💡 Instead of suppressing MagicNumber, use constants, the constant name provides context and acts as documentation as well. 

❓ Are we going with the custom spinner or using the usual ProgressBar everywhere for this feature?

Reply:
to follow up on the second question, there's spinners in Compose. If they're similar in the shopping feature, I already have one whipped up in a different PR that we can refactor into its own component.

https://github.com/mozilla-mobile/firefox-android/pull/3652

Reply:
As Kate commented, we decided to go with the default `ProgressIndicator`. Thanks for linking the spinner you added Noah! I put it in `IndeterminateProgressIndicator` to be reused.